### ISSUE TITLE
Drag and Drop fails when you click and then attempt to drag

### ISSUE THREAD

**Issue by dstrenge** [bpmn-io/bpmn-js#1465]
**Describe the Bug**

Drag and Drop fails when you click and then attempt to drag

**Steps to Reproduce**

1. Open https://demo.bpmn.io/new
2. Click on "Create Start Event" (or any other item in the toolbar)
3. Mouse-down and attempt to drag
4. Bug:  The item is not dragged onto the canvas
5. Mouse-down and attempt to drag again
6. Expected behavior

The click handler appears to interfere with the drag start handler

**Environment**

Browser: any
OS: any
Library version: 8.6.2

**Comment by nikku** [comment]
Could you attach a screen capture that showcases the behavior?

**Comment by dstrenge** [comment]
Here is the video.  I click the tool, show that it is working, then go back to the tool and attempt to drag.  Can be reproduced without moving the mouse onto the canvas (after the click).  Video was shot of Chrome on Windows 10.

[

https://user-images.githubusercontent.com/5981293/123425247-e2615000-d58f-11eb-946d-46164a63bc14.mp4

](url)

**Comment by pinussilvestrus** [comment]
So you are trying to create an element from the palette via click + then drag, is this correct? It really seems that start dragging closes the create-on-click operation. I could imagine that this is annoying to some extend. 

Can you describe how likely this scenario is? Is it rather common that you first click and then immediately drag?

**Comment by dstrenge** [comment]
Yes, that is exactly it (apologies that the video does not show the button clicks).  

I can only guess at how likely it is, but I've had 4 people (including myself) indicate that they think drag/drop is broken when they switch tools and this seems to be the cause of that.  I wonder if there is a way to indicate that the operation is in progress when you click (but haven't left the toolbar) - so that people know they don't have to drag at that point?

**Comment by pinussilvestrus** [comment]
Thanks a lot for providing some details! We will see how to improve the situation. 

> I wonder if there is a way to indicate that the operation is in progress when you click (but haven't left the toolbar) - so that people know they don't have to drag at that point?

Do you have any suggestion or idea that comes to your mind?

**Comment by dstrenge** [comment]
I was thinking that if you could see the shape on top of the toolbar when you start the action, possibly that would help tell the user that the action is in progress.  Otherwise, if the tool selected had a 'highlight' when you started the action, maybe that would be enough?

Tried to illustrate both possibilities here.  One with a green selection if the action is in progress, but also showing the shape above the toolbar.  UX is not my forte!

![image](https://user-images.githubusercontent.com/5981293/124468074-ab94f200-dd66-11eb-8b70-29757679f967.png)

**Comment by nikku** [comment]
We'll see how we could potentially follow up on this one.
